packages/python/plotly/plotly/express/_core.py

```diff
@@ -1425,9 +1431,18 @@ def build_dataframe(args, constructor):
             # def __dataframe__(self, ...):
             #   if not some_condition:
             #     self.to_pandas(...)
-            if not hasattr(df_not_pandas, "to_pandas"):
+            if not (
+                hasattr(df_not_pandas, "to_pandas")
+                or hasattr(df_not_pandas, "toPandas")
+                or hasattr(df_not_pandas, "to_pandas_df")
+            ):
                 raise exc
-            args["data_frame"] = df_not_pandas.to_pandas()
+            if hasattr(df_not_pandas, "toPandas"):
+                args["data_frame"] = df_not_pandas.toPandas()
+            elif hasattr(df_not_pandas, "to_pandas_df"):
+                args["data_frame"] = df_not_pandas.to_pandas_df()
+            else:
+                args["data_frame"] = df_not_pandas.to_pandas()
```

Comment:
Not a big deal, but it would be a bit cleaner to invert this:
```py
if hasattr(df_not_pandas, "toPandas"):
    args["data_frame"] = df_not_pandas.toPandas()
elif hasattr(df_not_pandas, "to_pandas_df"):
    args["data_frame"] = df_not_pandas.to_pandas_df()
elif hasattr(df_not_pandas, "to_pandas"):
    args["data_frame"] = df_not_pandas.to_pandas()
else:
    raise exc
```